src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -109,6 +112,15 @@ void EventPipe::FinishInitialize()
 {
     STANDARD_VM_CONTRACT;
 
+    s_CanStartThreads = true;
+
+    while (s_rgDeferredEventPipeSessionIds.Size() > 0)
+    {
+        EventPipeSessionID id = s_rgDeferredEventPipeSessionIds.Pop();
+        EventPipeSession *pSession = reinterpret_cast<EventPipeSession*>(id);
+        pSession->StartStreaming();
```

Comment:
StartStreaming is expected to be called with the EventPipe lock held. I am guessing this asserts in chk build?

Reply:
yes, I meant to call `EventPipe::StartStreaming` here